---
layout: page
title: Lectures
menu: true
order: 2
---

All Lectures are Tu/Th 9:00-12:00 pm in Warren Lecture Hall 2015 (WLH 2015) ([Map](https://goo.gl/maps/yPnFEypdSCM2)). Clicking on the class topics below will take you to corresponding lecture notes, homework assignments, pre-class video screen-casts and required reading material.   


| \#  | Date | **Topics for Fall 2018** |
| :---: | -------- | :--- |
| 1 | Tu, 10/02 | [**Welcome to Bioinformatics**](#1)  <br> Course introduction, Leaning goals &amp; expectations, Biology is an information science, History of Bioinformatics, Types of data, Application areas and introduction to upcoming course segments, Hands on with major Bioinformatics databases and key online NCBI and EBI resources  | 
|   |   |   |
| 2 | Th, 10/04 | [**Sequence alignment fundamentals, algorithms and applications**](#2) <br> Homology, Sequence similarity, Local and global alignment, classic Needleman-Wunsch, Smith-Waterman and BLAST heuristic approaches, Hands on with dot plots, Needleman-Wunsch and BLAST algorithms highlighting their utility and limitations | 
|   |   |   |
| 3 | Tu, 10/09 | [**Advanced sequence alignment and database searching**](#3) <br> Detecting remote sequence similarity, Database searching beyond BLAST, Substitution matrices, Using PSI-BLAST, Profiles and HMMs, Protein structure comparisons | 
|   |   |   |
| 4 | Th, 10/11 | [**Bioinformatics data analysis with R**](#4) <br> Why do we use R for bioinformatics? R language basics and the RStudio IDE, Major R data structures and functions, Using R interactively from the RStudio console  | 
|   |   |   |
| 5 | Tu, 10/16 | [**Data exploration and visualization in R**](#5) <br> The exploratory data analysis mindset, Data visualization best practices, Using and customizing base graphics (scatterplots, histograms, bar graphs and boxplots), Building more complex charts with ggplot and rgl | 
|   |   |   |
| 6 | Th, 10/18 | [**Why, when and how of writing your own R functions**](#6) <br> The basics of writing your own functions that promote code robustness, reduce duplication and facilitate code re-use | 
|   |   |   |
| 7 | Tu, 10/23 | [**Bioinformatics R packages from CRAN and BioConductor**](#7) <br> Extending functionality and utility with R packages, Obtaining R packages from CRAN and BioConductor, Working with Bio3D for molecular data | 
|   |   |   |
| 8 | Th, 10/25 | [**Introduction to Machine Learning for Bioinformatics 1**](#8) <br> Unsupervised learning, K-means clustering, Hierarchical clustering, Heatmap representations. Dimensionality reduction, Principal Component Analysis (PCA)  | 
|   |   |   |
| 9 | Tu, 10/30 | [**Unsupervised Learning Mini-Project**](#9) <br> Longer hands-on session with unsupervised learning analysis of cancer cells further highlighting Practical considerations and best practices for the analysis and visualization of high dimensional datasets | 
|   |   |   |
| 10 | Th, 11/01 | **Project:** [**Find a gene assignment (Part 1)**](#10) <br> Principles of database searching, sequence analysis, structure analysis along with [**Hands-on with Git**](#10) <br> How to perform common operations with the Git version control system. We will also cover the popular social code-hosting platforms GitHub and BitBucket.  | 
|   |   |   |
| 11 | Tu, 11/06 | [**Structural Bioinformatics (Part 1)**](#11) <br> Protein structure function relationships, Protein structure and visualization resources, Modeling energy as a function of structure | 
|   |   |   |
| 12 | Th, 11/08 | [**Bioinformatics in drug discovery and design**](#12) <br> Target identification, Lead identification, Small molecule docking methods, Protein motion and conformational variants, Molecular simulation and drug optimization | 
|   |   |   |
| 13 | Tu, 11/13 | [**Genome informatics and high throughput sequencing (Part 1)**](#13) <br> Genome sequencing technologies past, present and future; Biological applications of sequencing, Analysis of variation in the genome, and gene expression; The Galaxy platform along with resources from the EBI & UCSC; Sample Galaxy RNA-Seq workflow with FastQC and Bowtie2 | 
|   |   |   |
| 14 | Th, 11/15 | [**Transcriptomics and the analysis of RNA-Seq data**](#14) <br> RNA-Seq aligners, Differential expression tests, RNA-Seq statistics, Counts and FPKMs and avoiding P-value misuse, Hands-on analysis of RNA-Seq data with R. <br> **N.B.** Find a gene assignment part 1 due today!| 
|   |   |   |
| 15 | Tu, 11/20 | [**Genome annotation and the interpretation of gene lists**](#15) <br> Gene finding and functional annotation, Functional databases KEGG, InterPro, GO ontologies and functional enrichment | 
|   |   |   |
| 16 | Th, 11/22 | [**Happy Thanksgiving!**](#16)  | 
|   |   |   |
| 17 | Tu, 11/27 | [**Biological network analysis**](#17) <br> Network based approaches for integrating and interpreting large heterogeneous high throughput data sets; Discovering relationships in 'omics' data; Network construction, manipulation, visualization and analysis; Major graph theory and network topology measures and concepts (Degree, Communities, Shortest Paths, Centralities, Betweenness, Random graphs vs scale free); Hands-on with Cytoscape and igraph packages. | 
|   |   |   |
| 18 | Th, 11/29 | [**Cancer genomics**](#18) <br> Cancer genomics resources and bioinformatics tools for investigating the molecular basis of cancer. Mining the NCI Genomic Data Commons; Immunoinformatics and immunotherapy; Using genomics and bioinformatics to design a personalized cancer vaccine. Implications for personalized medicine. <br> **N.B.** Find a gene assignment due on Tuesday 03/13!  | 
|   |   |   |
| 19 | Tu, 12/04 | [**Course summary**](#19) <br> Summary of learning goals, Student course evaluation time and exam preparation; **Find a gene assignment due!** | 
|   |   |   |
| 20 | Th, 12/06 | [**Final exam!**](#20)  | 
|   |   |   | 

 


# Class material

<a name="1"></a>
## 1: Welcome to Bioinformatics and introduction to Bioinformatics databases and key online resources 

**Topics**:  
Course introduction, Leaning goals & expectations, Biology is an information science, History of Bioinformatics, Types of data, Application areas and introduction to upcoming course segments, Student 30-second introductions, Introduction to NCBI & EBI resources for the molecular domain of bioinformatics, Hands-on session using NCBI-BLAST, Entrez, GENE, UniProt, Muscle and PDB bioinformatics tools and databases. 

**Goals**: 
- Understand course scope, expectations, logistics and [ethics code]({{ site.baseurl }}/ethics/).  
- Understand the increasing necessity for computation in modern life sciences research.  
- Get introduced to how bioinformatics is practiced.  
- Complete the [pre-course questionnaire](https://goo.gl/forms/bSYYSXzxW7Pb0Qz73).  
- Setup your [laptop computer]({{ site.baseurl }}/setup/) for this course.  
- The goals of the hands-on session is to introduce a range of core bioinformatics databases and associated online services whilst actively investigating the molecular basis of several common human disease.

<!-- MAKE PRE-COURSE QUESTIONNAIRE and UPDATE LECTRUE SLIDES -->

**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture1-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture1-BIMM143-small.pdf){:.no-push-state},      
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture1-BIMM143_F18.pdf){:.no-push-state}  
- Feedback: [Muddy Point Assessment](https://goo.gl/forms/gMHFBsbkax64SN1a2){:.no-push-state},    
- Feedback: [Responses](https://docs.google.com/forms/d/e/1FAIpQLSemV6lwIbzeNA52gmKwfzLin63m-881j-DDHpgVekWJNtU4Rg/viewanalytics){:.no-push-state}.  

- Handout: [Class Syllabus]({{ site.baseurl }}/class-material/BIMM143_F18_syllabus.pdf){:.no-push-state}  
- Computer [Setup Instructions]({{ site.baseurl }}/setup/).  

**Homework**:  
- [Questions](https://goo.gl/forms/tS94rrJNvoqPsJKE2){:.no-push-state},  
- [Office Hours Sign Up](https://doodle.com/poll/pemmrpf49qygin26){:.no-push-state},  
- Readings: 
  - PDF1: [What is bioinformatics? An introduction and overview]({{ site.baseurl }}/class-material/bioinformatics_review.pdf){:.no-push-state},  
  - PDF2: [Advancements and Challenges in Computational Biology]({{ site.baseurl }}/class-material/bioinformatics_challenges_2015.pdf){:.no-push-state}.  
  

  
**Screen Casts**:  


<iframe width="560" height="315" src="https://www.youtube.com/embed/P2oSO7YPyfU?rel=0" frameborder="0" allowfullscreen></iframe>
**1 Welcome to BIMM-143:**
Course introduction and logistics.
{:.message}  

<br/>

<iframe width="560" height="315" src="https://www.youtube.com/embed/gJNXQfpErLY?rel=0" frameborder="0" allowfullscreen></iframe>
**2 What is Bioinformatics?**
Bioinformatics can mean different things to different people. What will we actually learn in this class?
{:.message}  

<br/>

<iframe width="560" height="315" src="https://www.youtube.com/embed/cCim7LrQZLY?rel=0" frameborder="0" allowfullscreen></iframe>
**3 How do we do Bioinformatics?**
Some basic bioinformatics can be done online or with downloaded tools. However, most often we will need a specialized computational setup.
{:.message}  


<hr>
<a name="2"></a>
## 2:  Sequence alignment fundamentals, algorithms and applications

**Topics**:  
Further coverage of *major NCBI & EBI resources* for the molecular domain of bioinformatics with a focus on GenBank, UniProt, Entrez and Gene Ontology. There are many bioinformatics databases (see [handout]({{ site.baseurl }}/class-material/Major_Databases_BIMM143.pdf){:.no-push-state}) and being able to judge their utility and quality is important. *Sequence Alignment and Database Searching*:
Homology, Sequence similarity, Local and global alignment, Heuristic approaches, Database searching with BLAST, E-values and evaluating alignment scores and statistics.
 

**Goals**:  
- Be able to query, search, compare and contrast the data contained in major bioinformatics databases (GenBank, GENE, UniProt, PFAM, OMIM, PDB) and describe how these databases intersect.  
- Be able to describe how nucleotide and protein sequence and structure data are represented (FASTA, FASTQ, GenBank, UniProt, PDB).
- Be able to describe how dynamic programming works for pairwise sequence alignment 
- Appreciate the differences between global and local alignment along with their major application areas.
- Understand how aligning novel sequences with previously characterized genes or proteins provides important insights into their common attributes and evolutionary origins. 
- The goals of the hands-on session are to explore the principles underlying the computational tools that can be used to compute and evaluate sequence alignments. 

**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture2-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture2-BIMM143-small.pdf){:.no-push-state},    
- [Handout: Major Bioinformatics Databases]({{ site.baseurl }}/class-material/Major_Databases_BIMM143.pdf){:.no-push-state}  
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture2-BIMM143_F18-lab.pdf){:.no-push-state}  
- Feedback: [Muddy Point Assessment](https://goo.gl/forms/rB1eOCK1qhwAVjwL2){:.no-push-state}.   
- Feedback: [Responses](https://docs.google.com/forms/d/e/1FAIpQLSfC84Hx43GAInuPaIF0kQJzj9pHqKJybZAzAGEasCu7lnIcSg/viewanalytics){:.no-push-state}.     


**Homework**:   
- [Quiz Questions](https://goo.gl/forms/0a6IL2nS7ifxd1nA2){:.no-push-state},  
- [Alignment Problem]({{ site.baseurl }}/class-material/lecture2-BIMM134_homework.pdf){:.no-push-state},  


**Readings**:    
- Readings: PDF1: [What is dynamic programming?]({{ site.baseurl }}/class-material/Dynamic_programming_primer.pdf){:.no-push-state},  
- Readings: PDF2 [Fundamentals of database searching]({{ site.baseurl }}/class-material/Fundamentals.pdf){:.no-push-state}.   

 
<hr>
<a name="3"></a>
<br>

## 3. Advanced sequence alignment and database searching

**Topics**:  
Detecting remote sequence similarity, Database searching beyond BLAST, PSI-BLAST, Profiles and HMMs, Protein structure comparisons, Beginning with command line based database searches.

**Goal**:  
- Be able to calculate the alignment score between two nucleotide or protein sequences using a provided scoring matrix
- Understand the limits of homology detection with tools such as BLAST 
- Be able to perform PSI-BLAST, HMMER and protein structure based database searches and interpret the results in terms of the biological significance of an e-value.
- Run our first bioinformatics tool from the command line.

**Material**:   
 
 - Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture3-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture3-BIMM143-small.pdf){:.no-push-state},    
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture3_BIMM143_F18-lab.pdf){:.no-push-state}  
- Feedback: [Muddy Point Assessment](https://goo.gl/forms/rgIo0DmNZlaioSdm1){:.no-push-state}.   
- Feedback: [Responses](https://docs.google.com/forms/d/e/1FAIpQLSdCuxK-O23Q31HdX1w0Bt7qM-6td8usRuVK1_8DkOBGDMI5GA/viewanalytics){:.no-push-state}.  



**Homework**:   
- [DataCamp Sign Up: see your UCSD email invite!](https://www.datacamp.com/enterprise/bimm143_f18-bioinformatics/){:.no-push-state},  
- [RStudio and R download and setup]({{ site.baseurl }}/setup/).  


<a name="4"></a>
<br>

## 4:  Bioinformatics data analysis with R
**Topics**: Why do we use R for bioinformatics? R language basics and the RStudio IDE, Major R data structures and functions, Using R interactively from the RStudio console.  

**Goal**: 
- Understand why we use R for bioinformatics
- Familiarity with R's basic syntax,  
- Be able to use R to read and parse comma-separated (.csv) formatted files ready for subsequent analysis,  
- Familiarity with major R data structures (vectors, matrices and data.frames),  
- Understand the basics of using functions (arguments, vectorizion and re-cycling).  

 
**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture4-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture4-BIMM143-small.pdf){:.no-push-state},  
- Lab: [Hands-on section 1]({{ site.baseurl }}/class-material/04_rintro/){:.no-push-state},  
- Feedback: [Muddy point assessment](https://goo.gl/forms/d23JGkVJyr0EcpTg2){:.no-push-state},  
- Feedback: [Responses](https://docs.google.com/forms/d/e/1FAIpQLSe9BUrwENHY3YNiEb-K6eQd8wxqAQ82e7XKCcrlF3ntj9OVZw/viewanalytics){:.no-push-state}.  


**Homework**:   
- [DataCamp Signup!](https://www.datacamp.com/enterprise/bimm143_f18-bioinformatics/assignments){:.no-push-state}.    



<hr>
<a name="5"></a>
<br>

## 5: Data exploration and visualization in R    
**Topics**: The exploratory data analysis mindset, Data visualization best practices, Simple base graphics (including scatterplots, histograms, bar graphs, dot chats, boxplots and heatmaps), Building more complex charts with ggplot.

**Goal**: 
- Appreciate the major elements of exploratory data analysis and why it is important to visualize data.
- Be conversant with data visualization best practices and understand how good visualizations optimize for the human visual system. 
- Be able to generate  informative graphical displays including scatterplots, histograms, bar graphs, boxplots, dendrograms and heatmaps and thereby gain exposure to the extensive graphical capabilities of R.
- Appreciate that you can build even more complex charts with ggplot and additional R packages such as rgl.



**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture5-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture5-BIMM143-small.pdf){:.no-push-state},  
- Rmarkdown documents for [plot session 1]({{ site.baseurl }}/class-material/05_draw_circle_points/){:.no-push-state}, and [more advanced plots]({{ site.baseurl }}/class-material/05_draw_lecture_plots2/){:.no-push-state},  
- Lab: [Hands-on worksheet]({{ site.baseurl }}/class-material/lecture5_lab_web.html){:.no-push-state}, 
- Lab: [PDF suplement]({{ site.baseurl }}/class-material/lecture5-BIMM143_lab.pdf){:.no-push-state}, 
- Example data for hands-on sections [bimm143_05_rstats.zip]({{ site.baseurl }}/class-material/bimm143_05_rstats.zip){:.no-push-state},  
- Feedback: [Muddy point assessment](https://goo.gl/forms/fYiwVbi7XsMKKGSS2){:.no-push-state},  
 
 
**Homework**:   
- This units homework is all [via **DataCamp** (see lecture 4 above)](https://www.datacamp.com/enterprise/bimm143_f18-bioinformatics/assignments){:.no-push-state}.   
 



<hr>
<a name="6"></a>
<br>

## 6: Why, When and How of Writing Your Own R Functions
**Topics**: , Using R scripts and Rmarkdown files, Import data in various formats both local and from online sources, The basics of writing your own functions that promote code robustness, reduce duplication and facilitate code re-use.

**Goals**:
- Be able to import data in various flat file formats from both local and online sources.
- Understand the structure and syntax of R functions and how to view the code of any R function.
- Understand when you should be writing functions.
- Be able to follow a step by step process of going from a working code snippet to a more robust function.

**Material**:
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture6-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture6-BIMM143-small.pdf){:.no-push-state},
- Flat files for importing with read.table: [test1.txt]({{ site.baseurl }}/class-material/test1.txt){:.no-push-state}, [test2.txt]({{ site.baseurl }}/class-material/test2.txt){:.no-push-state}, [test3.txt]({{ site.baseurl }}/class-material/test3.txt){:.no-push-state}.  
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture6-BIMM143_F18.pdf){:.no-push-state},  
- Feedback: [Muddy point assessment](https://goo.gl/forms/i5FWaVxlwb4Vxyim1){:.no-push-state},  


**Homework**:   
- See **Q6** of the [hands-on lab sheet above]({{ site.baseurl }}/class-material/lecture6-BIMM143_F18.pdf){:.no-push-state}. This entails turning a supplied code snippet into a more robust and re-usable function that will take any of the three listed input proteins and plot the effect of drug binding. Note assessment rubric and submission instructions within document. (Submission deadline: 9am **Th, 10/25**).    
- The remainder of this units homework is all [via **DataCamp**](https://www.datacamp.com/enterprise/bimm143_f18-bioinformatics/assignments){:.no-push-state}.   
 



<hr>
<a name="7"></a>
<br>

## 7: Using CRAN and Bioconductor Packages for Bioinformatics      
**Topics**: More on how to write R functions with worked examples. Further extending functionality and utility with R packages, Obtaining R packages from CRAN and Bioconductor, Working with Bio3D for molecular data, Managing genome-scale data with bioconductor.

**Goals**: 
- Be able to find and install R packages from CRAN and bioconductor,  
- Understand how to find and use package vignettes, demos, documentation, tutorials and source code repository where available.     
- Be able to write and (re)use basic R scripts to aid with reproducibility.  

**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture07-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture07-BIMM143-small.pdf){:.no-push-state},  
- [Collaborative Google Doc based notes on selected R packages](https://docs.google.com/document/d/1CDncv0mPBHoXpkQds0lD40sX32vM9EE_P01AG2MwzKo/edit?usp=sharing){:.no-push-state},    
- [Introductory tutorial on R packages](https://www.datacamp.com/community/tutorials/r-packages-guide){:.no-push-state},  
- Feedback: [Muddy point assessment](https://goo.gl/forms/wBXLNFE4iKMn1wv92){:.no-push-state}.      
- SideNote: [Convincing with graphics](https://xkcd.com/833/){:.no-push-state}.   
 
 
**Homework**:   
 See **Q6** from the last days [hands-on lab sheet above]({{ site.baseurl }}/class-material/lecture6-BIMM143_F18.pdf){:.no-push-state}. This entails turning a supplied code snippet into a more robust and re-usable function that will take any of the three listed input proteins and plot the effect of drug binding. Note assessment rubric and submission instructions within document. (Submission deadline: 9am **Th, 05/08**).    
- The remainder of this units homework is all [via **DataCamp**](https://www.datacamp.com/enterprise/bimm143_f18-bioinformatics/assignments){:.no-push-state}.   
 




<hr>
<a name="8"></a>
<br>

## 8: Introduction to Machine Learning for Bioinformatics    
**Topics**: Unsupervised learning, supervised learning and reinforcement learning; Focus on unsupervised learning, K-means clustering, Hierarchical clustering, Heatmap representations. Dimensionality reduction, visualization and analysis, Principal Component Analysis (PCA)
Practical considerations and best practices for the analysis of high dimensional datasets. 

**Goal**: 
- Understand the major differences between unsupervised and supervised learning.
- Be able to create k-means and hierarchical cluster models in R
- Be able to describe how the k-means and bottom-up hierarchical cluster algorithms work.
- Know how to visualize and integrate clustering results and select good cluster models.
- Be able to describe in general terms how PCA works and its major objectives. 
- Be able to apply PCA to high dimensional datasets and visualize and integrate PCA results (e.g identify outliers, find structure in features and aid in complex dataset visualization).



**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture08-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture08-BIMM143-small.pdf){:.no-push-state},  
- Lab: [Hands-on section worksheet for PCA]({{ site.baseurl }}/class-material/UK_food_pca//){:.no-push-state}  
- Data files: [UK_foods.csv]({{ site.baseurl }}/class-material/UK_foods.csv){:.no-push-state}.   
- [Introduction to PCA site](http://setosa.io/ev/principal-component-analysis/){:.no-push-state}. 
- Feedback: [Muddy point assessment](https://goo.gl/forms/4u6HDbPyS4al9JpJ3){:.no-push-state}.  

 



<hr>
<a name="9"></a>
<br>

## 9: Unsupervised Learning Mini-Project  
**Topics**: Longer hands-on session with unsupervised learning analysis of cancer cells, Practical considerations and best practices for the analysis and visualization of high dimensional datasets.  

**Goals**:   
- Be able to import data and prepare for unsupervised learning analysis.
- Be able to apply and test combinations of PCA, k-means and hierarchical clustering to high dimensional datasets and critically review results.

**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture09-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture09-BIMM143-small.pdf){:.no-push-state},  
- Lab: [Hands-on section worksheet for PCA]({{ site.baseurl }}/class-material/lecture9_lab.html){:.no-push-state}  
- Data file: [WisconsinCancer.csv]({{ site.baseurl }}/class-material/WisconsinCancer.csv){:.no-push-state}, [new_samples.csv]({{ site.baseurl }}/class-material/new_samples.csv){:.no-push-state}.     
- Bio3D PCA App: [http://bio3d.ucsd.edu/pca-app/](http://bio3d.ucsd.edu/pca-app/){:.no-push-state}.   
- Feedback: [Muddy point assessment](https://goo.gl/forms/1dkWdpJ4BNgtGHsR2){:.no-push-state}.   

- Bonus: [Kevin’s StackExchange Link on PCA](https://stats.stackexchange.com/questions/2691/making-sense-of-principal-component-analysis-eigenvectors-eigenvalues?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa){:.no-push-state}.   


<hr>
<a name="10"></a>
<br>


## 10: (**Project:**) Find a Gene Assignment Part 1        
The [**find-a-gene project**]({{ site.baseurl }}/class-material/Find_A_Gene_Project.pdf){:.no-push-state} is a required assignment for BIMM-143. The objective with this assignment is for you to demonstrate your grasp of database searching, sequence analysis, structure analysis and the R environment that we have covered to date in class.

You may wish to consult the scoring rubric at the end of the above linked project description and the [**example report**]({{ site.baseurl }}/class-material/Find_A_Gene_Project_Example.pdf){:.no-push-state} for format and content guidance.  
 
Your responses to questions Q1-Q4 are due at the beginning of class **Thursday Nov 15th** (11/15/18)).   

The complete assignment, including responses to all questions, is due at the beginning of class **Thursday Dec 4th** (12/04/18).  

Late responses will not be accepted under any circumstances.  

## Bonus: Hands-on with Git  
Today’s lecture and hands-on sessions introduce Git, currently the most popular version control system. We will learn how to perform common operations with Git and RStudio. We will also cover the popular social code-hosting platforms GitHub and BitBucket.  

- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture10-BIMM143_large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture10-BIMM143_small.pdf){:.no-push-state},  
- Lab: [Hands-on with Git](http://tinyurl.com/rclass-github){:.no-push-state}  



<hr>
<a name="11"></a>
<br>

## 11: Structural Bioinformatics    
**Topics**: Protein structure function relationships, Protein structure and visualization resources, Modeling energy as a function of structure, Homology modeling, Predicting functional dynamics, Inferring protein function from structure.

**Goal**: 
- View and interpret the structural models in the PDB,
- Understand the classic `Sequence>Structure>Function` via energetics and dynamics paradigm,
- Appreciate the role of bioinformatics in mapping the ENERGY LANDSCAPE of biomolecules,
- Be able to use the Bio3D package for exploratory analysis of protein sequence-structure-function-dynamics relationships.    


**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture11-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture11-BIMM143-small.pdf){:.no-push-state},  
- Lab: [Hands-on section worksheet for VMD and Bio3D]({{ site.baseurl }}/class-material/lecture11-BIMM143_F18.pdf){:.no-push-state}  
- Software links: [VMD download](http://www.ks.uiuc.edu/Development/Download/download.cgi){:.no-push-state},  [MUSCLE download](https://www.drive5.com/muscle/downloads.htm){:.no-push-state},  
- Feedback: [Muddy point assessment](https://goo.gl/forms/5eM3qYOhc2trBdUZ2){:.no-push-state}.   




<hr>
<a name="12"></a>
<br>

## 12: Bioinformatics in drug discovery and design 
**Topics**: The traditional path to drug discovery; High throughput screening  approaches; Computational receptor/target-based bioinformatics approaches; Computational ligand/drug-based bioinformatics approaches; Small molecule docking methods; Prediction and analysis of biomolecular complexes and drug optimization. 

**Goals**:    
- Appreciate how bioinformatics can aid drug discovery,  
- Be able to use Maestro for biomolecular visualization and analysis,
- Be able to perform *In silico* docking and virtual screening strategies for drug discovery,
- Understand the increasing role of bioinformatics in the drug discovery process.  


**Material**:   
- Software Download: [Schrödinger Discovery Suite](https://www.schrodinger.com/downloads/releases){:.no-push-state},  
- Software Product Key: **f4191b5b-56da-4a78-ad2d-0121e94c1827**   
- Pre-Class Survey: [Form One](https://goo.gl/forms/yy2lTwaBWNb7m9BY2){:.no-push-state},  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/Glide_VirtualScreening-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/Glide_VirtualScreening-small.pdf){:.no-push-state},  
- Lab: [Hands-on section worksheet for *In silico* drug docking]({{ site.baseurl }}/class-material/Visualization_Preparation_wHW.pdf){:.no-push-state},  
- Feedback: [Muddy point assessment](https://goo.gl/forms/mRd80EGeyXokyASg1){:.no-push-state},  
- Post-Class Survey: Form Two.  


**VMs**:  
- URL: [http://ucsdnov8.tsg.schrodinger.com](http://ucsdnov8.tsg.schrodinger.com){:.no-push-state},
- Login Deatails: [Username and passwords]({{ site.baseurl }}/class-material/Workshop-User-Accounts.pdf){:.no-push-state}.  


<hr>
<a name="13"></a>
<br>

## 13: Structural Bioinformatics II 
**Topics**: Revisiting bioinformatics approaches to drug discovery. Computational high throughput screening ; Receptor/target-based bioinformatics approaches; Computational ligand/drug-based bioinformatics approaches; Small molecule docking methods; Prediction and analysis of biomolecular motion, conformational variants and functional dynamics; Molecular simulation and drug optimization.

**Goals**:    
- Appreciate how bioinformatics can predict functional dynamics & further aid drug discovery,  
- Be able to apply open-source *In silico* docking and virtual screening strategies for drug discovery,
- Be able to use Bio3D for the analysis and prediction of protein flexibility,
- Understand the increasing role of bioinformatics in pharma and the drug discovery process in particular.  


**Material**:   
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture12B-BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture12B-BIMM143-small.pdf){:.no-push-state},
- Lab: [Hands-on section worksheet for *In silico* drug docking]({{ site.baseurl }}/class-material/lecture12B-BIMM143.pdf){:.no-push-state}
- Software download links: [AutoDock Tools](http://mgltools.scripps.edu/downloads){:.no-push-state},  [AutoDock Vina](http://vina.scripps.edu/download.html){:.no-push-state},  [VMD](http://www.ks.uiuc.edu/Development/Download/download.cgi){:.no-push-state},  [MUSCLE](https://www.drive5.com/muscle/downloads.htm){:.no-push-state}
- Optional backup files: [config.txt]({{ site.baseurl }}/class-material/config.txt){:.no-push-state}, [1hsg_protein.pdbqt]({{ site.baseurl }}/class-material/1hsg_protein.pdbqt){:.no-push-state}, [ligand.pdbqt]({{ site.baseurl }}/class-material/ligand.pdbqt){:.no-push-state}, [log.txt]({{ site.baseurl }}/class-material/log.txt){:.no-push-state}, [all.pdbqt]({{ site.baseurl }}/class-material/all.pdbqt){:.no-push-state}
- Feedback: [Muddy point assessment](https://goo.gl/forms/pFcCQlDVzIClxI8R2){:.no-push-state}.  




<hr>
<a name="14"></a>
<br>

## 14: Genome informatics and high throughput sequencing (Part 1) 
**Topics**: Genome sequencing technologies past, present and future (Sanger, Shotgun, PacBio, Illumina, toward the $500 human genome), Biological applications of sequencing, Variation in the genome, RNA-Sequencing for gene expression analysis; Major genomic databases, tools and visualization resources from the EBI & UCSC, The Galaxy platform for quality control and analysis; Sample Galaxy RNA-Seq workflow with FastQC and Bowtie2

**Goals**: 
- Appreciate and describe in general terms the rapid advances in sequencing technologies and the new areas of investigation that these advances have made accessible.  
- Understand the process by which genomes are currently sequenced and the bioinformatics processing and analysis required for their interpretation.  
- For a genomic region of interest (e.g. the neighborhood of a particular SNP), use a genome browser to view nearby genes, transcription factor binding regions, epigenetic information, etc.
- Be able to use the Galaxy platform for basic RNA-Seq analysis from raw reads to expression value determination.  
- Understand the FASTQ file format and the information it holds. 
- Understand the [SAM/BAM file format]({{ site.baseurl }}//class-material/sam_format/){:.no-push-state} and the information it holds.  


**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture14_BIMM143_F18-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture14_BIMM143_F18-small.pdf){:.no-push-state},
- [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture14-BIMM143_F18-lab.pdf){:.no-push-state},
- RNA-Seq data files: [HG00109_1.fastq]({{ site.baseurl }}/class-material/HG00109_1.fastq){:.no-push-state}, [HG00109_2.fastq]({{ site.baseurl }}/class-material/HG00109_2.fastq){:.no-push-state}, [genes.chr17.gtf]({{ site.baseurl }}/class-material/genes.chr17.gtf){:.no-push-state}, [Expression genotype results]({{ site.baseurl }}/class-material/rs8067378_ENSG00000172057.6.txt){:.no-push-state}, [Example R script]({{ site.baseurl }}/class-material/lecture13_plot.r){:.no-push-state}, [Example Rmd](https://github.com/bioboot/test_github/blob/master/lecture13_plot.md){:.no-push-state}.
- [SAM/BAM file format description]({{ site.baseurl }}//class-material/sam_format/){:.no-push-state}.  
- Feedback: [Muddy point assessment](https://goo.gl/forms/cr9Nxn819FDbrNtU2){:.no-push-state}.  

## IPs     
- 149.165.157.76  
- 149.165.156.175  
- 149.165.168.14  
- 129.114.17.25  
- 149.165.168.58  
- 129.114.17.186
- 149.165.168.198
- 129.114.104.88
- 129.114.104.18  
- 149.165.168.123
- 149.165.157.208
- 149.165.156.158
- 129.114.104.32
- 149.165.157.44
- 149.165.168.199
- 149.165.168.217
- 149.165.168.216
- 129.114.16.25
- 149.165.168.212
- 129.114.16.27
- to be updated


<hr>
<a name="15"></a>
<br>

## 15: Transcriptomics and the analysis of RNA-Seq data
**Topics**: Analysis of RNA-Seq data with R, Differential expression tests, RNA-Seq statistics, Counts and FPKMs, Normalizing for sequencing depth, DESeq2 analysis.

**Goals**:
- Given an RNA-Seq dataset, find the set of significantly differentially expressed genes and their annotations.
- Gain competency with data import, processing and analysis with DESeq2 and other bioconductor packages.
- Understand the structure of count data and metadata required for running analysis.
- Be able to extract, explore, visualize and export results.

**Material**:
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture14_BIMM143_W18-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture14_BIMM143_W18-small.pdf){:.no-push-state}.
- Detailed [Bioconductor setup]({{ site.baseurl }}//class-material/bioconductor_setup/){:.no-push-state} instructions.
- [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture14-BIMM143_W18-lab/){:.no-push-state}
- Data files: [airway_scaledcounts.csv]({{ site.baseurl }}/class-material/airway_scaledcounts.csv){:.no-push-state}, [airway_metadata.csv]({{ site.baseurl }}/class-material/airway_metadata.csv){:.no-push-state}, [annotables_grch38.csv]({{ site.baseurl }}/class-material/annotables_grch38.csv){:.no-push-state}.
- Muddy point assessment


**Readings**:
 - Excellent review article: [Conesa et al. A survey of best practices for RNA-seq data analysis. _Genome Biology_ 17:13 (2016)](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8){:.no-push-state}.
 - An oldey but a goodie: [Soneson et al. "Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences." _F1000Research_ 4 (2015)](https://f1000research.com/articles/4-1521/v2).
 - Abstract and introduction sections of: [Himes et al. "RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells." _PLoS ONE_ 9.6 (2014): e99625](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099625){:.no-push-state}.




<hr>
<a name="16"></a>
<br>


## 16: Genome annotation and the interpretation of gene lists 
**Topics**: Gene finding and functional annotation from high throughput sequencing data, Functional databases KEGG, InterPro, GO ontologies and functional enrichment

**Goals**: Perform a GO analysis to identify the pathways relevant to a set of genes (e.g. identified by transcriptomic study or a proteomic experiment). Use both Bioconductor packages and online tools to interpret gene lists and annotate potential gene functions.

**Material**:   
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture15_BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture15_BIMM143-small.pdf){:.no-push-state}.  
- [Hands-on section worksheet]({{ site.baseurl }}/class-material/lecture15_BIMM143_W18/){:.no-push-state}  
- Data files: [GSE37704_featurecounts.csv]({{ site.baseurl }}/class-material/GSE37704_featurecounts.csv){:.no-push-state}, [GSE37704_metadata.csv]({{ site.baseurl }}/class-material/GSE37704_metadata.csv){:.no-push-state}.   
- Muddy point assessment
 

**R Knoweledge Check**:
[**Quiz Assessment**](https://goo.gl/forms/QiBFMBZhcHF9ZlU83){:.no-push-state} 

**Readings**:    
 - Good review article: Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "*Differential analysis of gene regulation at transcript resolution with RNA-seq*". Nat Biotechnol 2013 Jan;31(1):46-53. [PMID: 23222703](https://www.ncbi.nlm.nih.gov/pubmed/23222703){:.no-push-state}.  

  
  




<!--

## 16: Essential statistics for bioinformatics 
**Topics**: Data summary statistics; Inferential statistics; Significance testing; Two sample T-test in R; Power analysis in R; Multiple testing correction; and almost everything you wanted to know about Principal Component Analysis (PCA) but were afraid to ask! Extensive R examples and applications.

**Material**:   
- Lecture Slides: [PDF]({{ site.baseurl }}/class-material/lecture16_BIMM143_W18-large.pdf){:.no-push-state}.  
- Data files: 
  - [RNA-seq_excitatory_neurons_rep1_TPM_by_names.tsv]({{ site.baseurl }}/class-material/RNA-seq_excitatory_neurons_rep1_TPM_by_names.tsv){:.no-push-state}, 
  - [RNA-seq_excitatory_neurons_rep2_TPM_by_names.tsv]({{ site.baseurl }}/class-material/RNA-seq_excitatory_neurons_rep2_TPM_by_names.tsv){:.no-push-state}, 
  - [RNA-seq_PV_neurons_rep1_TPM_by_names.tsv]({{ site.baseurl }}/class-material/RNA-seq_PV_neurons_rep1_TPM_by_names.tsv){:.no-push-state}, 
  - [RNA-seq_PV_neurons_rep2_TPM_by_names.tsv]({{ site.baseurl }}/class-material/RNA-seq_PV_neurons_rep2_TPM_by_names.tsv){:.no-push-state}.   
- Feedback: [Muddy point assessment](https://goo.gl/forms/YBPM7MoMTqKNnLoV2){:.no-push-state}.  

-->




<hr>
<a name="17"></a>
<br>


## 17: Biological network analysis 
**Topics**: Network graph approaches for integrating and interpreting large heterogeneous high throughput data sets; Discovering relationships in 'omics' data; Network construction, manipulation, visualization and analysis; Graph theory; Major network topology measures and concepts (Degree, Communities, Shortest Paths, Centralities, Betweenness, Random graphs vs scale free); De novo sub-network construction and clustering. Hands-on with Cytoscape and R packages for network visualization and analysis.

**Goals**:
- Be able to describe the major goals of biological network analysis and the concepts underlying network visualization and analysis.   
- Be able to use Cytoscape for network visualization and manipulation.
- Be able to find and instal Cytoscape Apps to extend network analysis functionality.  
- Appreciate that the igraph R package has extensive network analysis functionality beyond that in Cytoscape and that the R bioconductor package RCy3 package allows us to bring networks and associated data from R to Cytoscape so we can have the best of both worlds. 

**Material**:  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/lecture17_BIMM143-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/lecture17_BIMM143-small.pdf){:.no-push-state}.  
- [Hands-on section worksheet Part 1.]({{ site.baseurl }}/class-material/lecture17_BIMM143_S18_lab1.pdf){:.no-push-state}  
- [Hands-on section worksheet Part 2.]({{ site.baseurl }}/class-material/lecture17_BIMM143_W18/){:.no-push-state}  
- Data files:  
  - [galFiltered.sif]({{ site.baseurl }}/class-material/galFiltered.sif){:.no-push-state},  
  - [galExpData.csv]({{ site.baseurl }}/class-material/galExpData.csv){:.no-push-state},  
  - [CytoscapeDemo_01.cys]({{ site.baseurl }}/class-material/CytoscapeDemo_01.cys){:.no-push-state},  
  - [virus_prok_cor_abundant.tsv]({{ site.baseurl }}/class-material/virus_prok_cor_abundant.tsv){:.no-push-state},  
  - [phage_ids_with_affiliation.tsv]({{ site.baseurl }}/class-material/phage_ids_with_affiliation.tsv){:.no-push-state}, 
  - [prok_tax_from_silva.tsv]({{ site.baseurl }}/class-material/prok_tax_from_silva.tsv){:.no-push-state}.  
- Muddy point assessment
 

  

<hr>
<a name="18"></a>
<br>

## 18: Cancer genomics 
**Topics**: Cancer genomics resources and bioinformatics tools for investigating the molecular basis of cancer. Large scale cancer sequencing projects; NCI Genomic Data Commons; What has been learned from genome sequencing of cancer? **Immunoinformatics, immunotherapy  and cancer**; Using genomics and bioinformatics to harness a patient’s own immune system to fight cancer. Implications for the development of personalized medicine.

**Material**: 
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/Lecture18_BIMM143_S18-large.pdf){:.no-push-state}, [Small PDF]({{ site.baseurl }}/class-material/Lecture18_BIMM143_S18-small.pdf){:.no-push-state}.  
- [Hands-on section worksheet Part 1.]({{ site.baseurl }}/class-material/lecture18_part1_BIMM143_W18/){:.no-push-state}  
- [Hands-on section worksheet Part 2.]({{ site.baseurl }}/class-material/lecture18_part2_BIMM143_W18/){:.no-push-state}  
- Data files:  
  - [lecture18_sequences.fa]({{ site.baseurl }}/class-material/lecture18_sequences.fa){:.no-push-state},  
- Solutions:  
  - Example [mutant identification and subsequence extraction with R]({{ site.baseurl }}/class-material/lecture18_part2_example/){:.no-push-state} walk through.    
  - [subsequences.fa]({{ site.baseurl }}/class-material/subsequences.fa){:.no-push-state},  
  - [Solutions.pdf]({{ site.baseurl }}/class-material/Solutions.pdf){:.no-push-state}.  
- IEDB HLA binding prediction website [http://tools.iedb.org/mhci/](http://tools.iedb.org/mhci/){:.no-push-state}.  

<hr>
<a name="19"></a>
<br>

## 19: Course summary  
**Topics**: Summary of learning goals, Student course evaluation time and exam preparation; Find a gene assignment due.  Open study.  
  
Hand-out: [**Exam guidlines, topics, and example questions**]({{ site.baseurl }}/class-material/BIMM143_exam_guidlines.pdf){:.no-push-state}

Ether-pad: [**Feedback**](https://etherpad.net/p/bimm143_S18){:.no-push-state}.  
  
  
<hr>
<a name="20"></a>
<br>

## 20: Final Exam  

This open-book, open-notes 150-minute test consists of 35 questions.  The number of points for each question is indicated in green font at the beginning of each question. There are 80 total points on offer.  


Please remember to:

- Read all questions carefully before starting.
- Put your name, UCSD email and PID number on your test.
- Write all your answers on the space provided in the exam paper.
- Remember that concise answers are preferable to wordy ones.
- Clearly state any simplifying assumptions you make in solving a problem.
- No copies of this exam are to be removed from the class-room.
- No talking or communication (electronic to otherwise) with your fellow students once the exam has begun.
- **Good luck!**  


<hr>
<a name="21"></a>
<br>

## 21: Bonus: Bioinformatics & Genomics in industry  

Friday Dec 7th at 1pm come and enjoy a set of short open ended guest lectures from leading genomic scientists at Illumina Inc., Synthetic Genomics Inc., and the La Jolla Institute for Allergy and Immunology. Come prepared for networking and to have your questions about industry careers in Genomics answered. 

